setup.py

```diff
@@ -119,6 +119,13 @@ def run(self):
         sdist.run(self)
 
 
+efilter_version = __version__
+if "bdist_msi" in sys.argv and efilter_version[1] == "!":
+  # bdist_msi does not support a setuptools epoch version so we remove it
+  # as a work around.
+  _, _, efilter_version = efilter_version.partition("!")
```

Comment:
Im not sure that bdist_msi is still supported, although it is there (as you say it does not support the current versioning scheme properly). Unless you have a specific use case I would vote not to support it particularly.

The downside is that you might expect bug reports for something which is not our responsibility to support and that noone will use.